core/trino-spi/src/main/java/io/trino/spi/expression/Call.java

```diff
@@ -57,6 +59,20 @@ public ConnectorExpression getArgument(int index)
         return arguments.get(index);
     }
 
+    public <T extends ConnectorExpression> T getArgument(int index, Class<T> clazz)
```

Comment:
I don't think we should have such an API. a cast at call site is as verbose, and this method doesn't provide any more type safety than a cast.